### PR TITLE
refactor: iteration 1 of starting to refactor doSync code before new data model

### PR DESCRIPTION
**Pull request by albertyfwu**

### CODE REVIEW

**Review by tomkit — APPROVED** [review]

**Comment by tomkit** [comment]
cc @khennes

**Comment by khennes** [comment]
> cc @khennes

yup took a look earlier, thanks!

File: apps/api/temporal/activities/do_sync.ts [hunk 1]
```diff
@@ -172,19 +119,20 @@ async function writeRecordsToDestination({
...
+  }: {
+    destination: PostgresDestination;
+    fieldMapping: Record<string, string>;
+    mappedRecords: Record<string, unknown>[];
+    customerId: string;
+  }
+): Promise<void> {
   // TODO: What do we do if there are columns missing in the source?
   // TODO: Check that there are actually columns to sync over
   const { upsertKey } = destination.config;
```
Comment (tomkit) [inline apps/api/temporal/activities/do_sync.ts:138]: nit: could these be put until a util to improve readability

**Review by tomkit — APPROVED** [review]

File: apps/api/temporal/activities/do_sync.ts (hunk 1 above)
Comment (albertyfwu) [reply apps/api/temporal/activities/do_sync.ts:138]: maybe i can do this in a later PR. every new commit on approval now requires new approval

**Pull request merged**